### ISSUE TITLE
[Persistence] Reconcile different approaches to failure notification

### ISSUE THREAD

**Issue by akhenry** [nasa/openmct#373]
Currently there are three different mechanisms for handling errors on persistence. 
1. The `PersistenceFailureHandler` notifies the user of errors when the persistence queue is enabled via a blocking dialog.
2. The `PersistenceCapability` provides blanket notification of all persistence failures for all providers via a non-blocking notification.
3. In at least one instance (the `CopyAction`) feature-specific error handling is implemented that detects errors bubbled from the persistence provider and shows a blocking error message using the showBlockingMessage method on the DialogService.

These three approaches should be consolidated into one consistent approach for notifying the user of persistence failures.

Based on discussions in #58 the preferred approach is **3.** ( @VWoeltjen I'd like to hear some more of your thoughts on this), however this is inconsistent with **1**. Additionally, **2.** is not ideal as it mixes UI concerns into the `CapabilityProvider` and pre-empts more context appropriate user feedback that should happen at a higher level.

**Comment by akhenry** [comment]
>I think we have an alternate issue for removing the persistence queue; I think the persistence service doing anything more than resolve/reject is out of scope. The service shouldn't have UI bindings.

>I would expect individual operations to handle persistence failures differently as the failure of each user-action could potentially have different UX requirements, and each failure may require different clean up in the case of batch operations.

>At baseline: implement support for individual failures (e.g. copy fail, save fail, link fail) instead of a global persistence failure handler. Once that is done, if we see a common pattern, then we support that, but from experience there isn't a global pattern that provides good UX without significant effort.
@larkin

**Comment by akhenry** [comment]
> I agree that the persistence errors should be bubbled up so that context appropriate handling can take place. The generic persistence error handler is proposed as an alternative that integrates with and extends the current persistence queue error handling.

> There may be a case for generic handling of certain types of errors, or for abstracting details of the underlying persistence store. If we want to give users the option to overwrite or discard conflicting modifications for example, implementing this separately for copy, save, link, etc. doesn't make much sense. But in any case, any generic handling should happen higher in the stack.

**Comment by akhenry** [comment]
As an additional note, currently error on copy causes a notification message for every item being copied. This is due to persistence errors being triggered in the persistence handler, which is too low in the stack. Persistence errors should bubble. Duplication error should be reported once.

**Comment by larkin** [comment]
there are several related issues that we should be merging.
